src/buildercore/bootstrap.py

```diff
@@ -104,7 +110,9 @@ def _create_generic_stack(stackname, parameters=None, on_start=_noop, on_error=_
         LOG.info("Stack taking a long time to complete: %s", err)
         raise
 
-    except BotoServerError as err:
+    except botocore.exceptions.ClientError as err:
+        # TODO: is this ' already exists' check different from the other 'does not exist' check?
+        # if so, can we get a list of these messages somewhere?
         if err.message.endswith(' already exists'):
```

Comment:
this has changed to:
```
>>> conn.create_stack(StackName='medium--ci', TemplateBody='{}')
INFO - boto3.resources.action - Calling cloudformation:create_stack with {'StackName': 'medium--ci', 'TemplateBody': '{}'}
2018-04-20 16:09:59,376 botocore.vendored.requests.packages.urllib3.connectionpool [INFO] Starting new HTTPS connection (1): cloudformation.us-east-1.amazonaws.com
INFO - botocore.vendored.requests.packages.urllib3.connectionpool - Starting new HTTPS connection (1): cloudformation.us-east-1.amazonaws.com
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/giorgio/code/builder/venv/local/lib/python2.7/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/home/giorgio/code/builder/venv/local/lib/python2.7/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/home/giorgio/code/builder/venv/local/lib/python2.7/site-packages/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/giorgio/code/builder/venv/local/lib/python2.7/site-packages/botocore/client.py", line 599, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.AlreadyExistsException: An error occurred (AlreadyExistsException) when calling the CreateStack operation: Stack [medium--ci] already exists
```

Reply:
so it should be easier to catch

Reply:
it has some issues: https://github.com/boto/boto3/issues/1195

it says to use botocore.exceptions but it doesn't exist there either. It is bypassing the adhoc 'exists' check on the message string rather than returning false, so good catch